### PR TITLE
add HPC-UGent specific chapter on (pilot) usage of GPGPU cluster

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: intro-HPC/ch_gpu_gent.tex [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+%\shellcmd{module swap cluster/.joltik}
+\end{prompt}
+
+Please note the dot (\lstinline|.|) in the module name (which makes this \lstinline|cluster| module a hidden module).
+
+Then use the familiar \lstinline|qsub|, \lstinline|qstat|, etc.\ commands, taking into account the guidelines outlined
+in section~\ref{sec:gpu_ugent_resources}.
+
+\strong{Note that you will only be able to submit jobs to the \lstinline|joltik| cluster if you are a member of the
+pilot group (see \ref{sec:gpu_ugent_pilot}).} If not, you will see the following error message:
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: the `.` is part of the `\ref`

File: intro-HPC/ch_gpu_gent.tex [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+
+\begin{prompt}
+salloc: error:
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: maybe remove this first part of the message?

File: intro-HPC/ch_gpu_gent.tex [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+\label{sec:gpu_ugent_resources}
+
+There are 2 main ways to ask for GPUs as part of a job:
+
+\begin{itemize}
+  \item Either as a node property (similar to the number of cores
+  per node specified via \lstinline|ppn|) using \lstinline|-l nodes=X:ppn=Y:gpus=Z| (where the \lstinline|ppn=Y| is optional),
+  or as a separate resource request (similar to the amount of memory) via \lstinline|-l gpus=Z|.
+  Both notations give exactly the same result.
+  The \lstinline|-l gpus=Z| is convenient is you only need one
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: i'll try to fix the `nodes=1:ppn=1` issue from th esubmitfilter asap

File: intro-HPC/ch_gpu_gent.tex [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+  node and you are fine with the default number of cores per GPU.
+  The \lstinline|-l nodes=...:gpus=Z| notation is required if you want to run with full control or in multinode cases like MPI jobs.
+  In you do not specify the number of GPUs, you get by default 1 GPU.
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: euhm, users still have to ask the `-l gpus` or `:gpus` part. it's not that you always get a gpu by default (or am i reading too much into this?)

File: intro-HPC/ch_gpu_gent.tex [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+% Attention points %
+%%%%%%%%%%%%%%%%%%%%
+
+\section{Attention points}
+\label{sec:gpu_ugent_attention_points}
+
+Some important attention points:
+
+\begin{itemize}
+\item For MPI jobs, we recommend the (new) wrapper \lstinline|mypmirun| from the \lstinline|vsc-mympirun| module
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: was there a specific version or it is part of the latest/default version?

File: intro-HPC/ch_gpu_gent.tex [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+  are spread heterogenous over nodes from using the \lstinline|--gpus Z| option).
+\item Both \lstinline|mypmirun| and \lstinline|wurker| will try to do the most optimised placement of cores and tasks, and
+  will provide 1 (optimal) GPU per task/MPI rank, and set one so-called \emph{visible device}
+  (i.e. \lstinline|CUDA_VISIBLE_DEVICES| only has 1 ID). The actual devices are not constrained to the ranks,
+  so you can access all devices requested in the job.
+  \emph{We know that at this moment, this is not working properly, but we are working on this. We advise against trying to fix this yourself.}
+  % TODO: this is still not the case, due to bugs in slurm. For now, you will probably do not get optimal placement and/or more
+  %than one visible device.
+  %TODO: we need an easy way to toggle this behaviour.
+\item Interactive jobsubmission with GPUs will result in a shell that has no access to the requested GPUs. To get access, you need to use
```
Comment (stdweird) [inline intro-HPC/ch_gpu_gent.tex]: remove it

File: intro-HPC/examples/Using-the-GPU-cluster-at-HPC-UGent/TensorFlow_GPU_joltik.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+#PBS -l walltime=1:0:0
+#PBS -l nodes=1:ppn=32,gpus=4
+
+module load PyTorch/1.2.0-fosscuda-2019.08-Python-3.7.2
```
Comment (stdweird) [inline intro-HPC/examples/Using-the-GPU-cluster-at-HPC-UGent/TensorFlow_GPU_joltik.sh]: maybe the example should only use one of TF or pytorch?

File: intro-HPC/ch_gpu_gent.tex (hunk 1 above)
Comment (boegel) [reply intro-HPC/ch_gpu_gent.tex]: No, it's part of the `\strong{}`?

Comment (stdweird) [reply intro-HPC/ch_gpu_gent.tex]: ouch, a bit too tired i guess

File: intro-HPC/ch_gpu_gent.tex (hunk 4 above)
Comment (boegel) [reply intro-HPC/ch_gpu_gent.tex]: You wrote this, not me. ;)

I'll reword it.

File: intro-HPC/ch_gpu_gent.tex (hunk 5 above)
Comment (boegel) [reply intro-HPC/ch_gpu_gent.tex]: They should always load default/latest (there's only one version installed on `joltik` now, the `5.0.0-beta`)

File: intro-HPC/examples/Using-the-GPU-cluster-at-HPC-UGent/TensorFlow_GPU_joltik.sh (hunk 7 above)
Comment (boegel) [reply intro-HPC/examples/Using-the-GPU-cluster-at-HPC-UGent/TensorFlow_GPU_joltik.sh]: Yeah, that was meant as an example for you yo tweak.

I'll make it load just one module, but this isn't actually included in the documentation currently.

I can add it back though...

**Pull request merged**